src/internal/util/tetris.ts

```diff
@@ -0,0 +1,51 @@
+export class Tetris<T extends any> {
```

Comment:
Based off of the name, I'm not sure what the intent of this class is. Why is it called "Tetris"?

Reply:
@benlesh 

Apologies for the poor documentation. I wasn't sure if this change would be accepted so I just over-wrote a  deprecated operator for now, to demonstrate the change. It should, of-course, be renamed to something more appropriate and the code broken down and moved to a different folder (Happy to do it).

**Setup**: On subscription to 2 observables (which have the replay(buffer: 5) and share operators) the first one emits 4 values (a, b, c, d) and the second one emits 1 (e).
**Expectation**: Since both the hot observables have values ready to be forwarded, the expectation was that combineLatest would give (ae, be, ce, de).
**Actual**: The received value is (de).

The operator is similar to combineLatestAll, but differs from it by storing all the values that it receives from the time of subscription in the Tetris object.
Consider the example in the [description](https://github.com/ReactiveX/rxjs/pull/6769#issue-1103188742). The combineLatestAll operator would have just emitted the combination `cd`. But this operator stored the values emitted by the first observable until the second one emitted its first, following which it sequentially emitted all the values from the first observable paired with the one value of the second.

The purpose of the Tetris object is:
 - to maintain one array of emitted values per observable (column value in constructor)
 - to keep track of the last emitted values (lastEmittedIndices)
 - to let us know if there is a new available value (hasNext)
 - return an array of most recently emitted values, in case of a new emission (getNext)

I chose the name Tetris because I played a lot of it when I was a kid. When I saw how the values were "falling" and "disappearing" in the columns, it reminded me of it. It can/should be renamed to something more appropriate that betrays its intent a bit more clearly.

Reply:
I see. So what you're trying to solve here is the condition where the first values of the combined observables all show up synchronously, and you might miss a few values from the "earlier" observables while waiting for the subsequent values to arrive? 

I'm not sure if we can land this in RxJS core, as it's so similar to combineLatest, and probably confusing. I've seen requests for things like this before, but usually it's for people wanting this:

```
a$:   ---1----2-----4-----5-----
b$;   ------------3----------6----
out$: ---A----B---C--D-----E--F--- 

A: [1, null]
B: [2, null]
C: [2, 3]
D: [4, 3]
E: [5, 3]
F: [5, 6]
```

And I *think* that operator already exists in @cartant's rxjs-etc library.

Reply:
<pre><code>
a$:   ---1----2-----------4-----5-----
b$;   ------------3----------6----
out$: ------------(AB)----C--D--E--- 

A: [1, 3]
B: [2, 3]
C: [4, 3]
D: [4, 6]
E: [5, 6]
</pre></code>

Reply:
@benlesh The example posted in your comment doesn't reflect the way this operator works. Can you please recheck the expected output I have posted and confirm if this operator indeed exists in rxjs-etc?